ppdet/modeling/backbone/resnet.py

```diff
@@ -0,0 +1,276 @@
+                padding=0,
+                act=None,
+                learning_rate=learning_rate)
+
+        self.conv1 = ConvBNLayer(
+            name_scope + "_branch2a",
+            ch_in=ch_in,
+            ch_out=ch_out,
+            filter_size=1,
+            stride=stride,
+            padding=0,
+            learning_rate=learning_rate, )
+
+        self.conv2 = ConvBNLayer(
+            name_scope + "_branch2b",
+            ch_in=ch_out,
+            ch_out=ch_out,
+            filter_size=3,
+            stride=1,
+            padding=1,
+            learning_rate=learning_rate)
+
+        self.conv3 = ConvBNLayer(
+            name_scope + "_branch2c",
+            ch_in=ch_out,
+            ch_out=ch_out * 4,
+            filter_size=1,
+            stride=1,
+            padding=0,
+            learning_rate=learning_rate,
+            act=None)
+        self.name_scope = name_scope
+
+    def forward(self, inputs):
+        if self.shortcut:
+            short = inputs
+        else:
+            short = self.short(inputs)
+
+        conv1 = self.conv1(inputs)
+        conv2 = self.conv2(conv1)
+        conv3 = self.conv3(conv2)
+
+        out = fluid.layers.elementwise_add(
+            x=short,
+            y=conv3,
+            act='relu',
+            name=self.name_scope + ".add.output.5")
+
+        return out
+
+
+class Blocks(Layer):
+    def __init__(self,
+                 name_scope,
+                 ch_in,
+                 ch_out,
+                 count,
+                 stride,
+                 learning_rate=1.0):
+        super(Blocks, self).__init__()
+
+        self.blocks = []
+        for i in range(count):
+            if i == 0:
+                name = name_scope + "a"
+                self.stride = stride
+                self.shortcut = False
+            else:
+                name = name_scope + chr(ord("a") + i)
+                self.stride = 1
+                self.shortcut = True
+
+            block = self.add_sublayer(
+                name,
+                BottleNeck(
+                    name,
+                    ch_in=ch_in if i == 0 else ch_out * 4,
+                    ch_out=ch_out,
+                    stride=self.stride,
+                    shortcut=self.shortcut,
+                    learning_rate=learning_rate))
+            self.blocks.append(block)
+            shortcut = True
+
+    def forward(self, inputs):
+        res_out = self.blocks[0](inputs)
+        for block in self.blocks[1:]:
+            res_out = block(res_out)
+        return res_out
+
+
+@register
+@serializable
+class ResNet(Layer):
+    def __init__(
+            self,
+            norm_type='bn',
+            depth=50,
+            feature_maps=4,
+            freeze_at=2, ):
+        super(ResNet, self).__init__()
+
+        if depth == 50:
+            blocks = [3, 4, 6, 3]
+        elif depth == 101:
+            blocks = [3, 4, 23, 3]
+        elif depth == 152:
+            blocks = [3, 8, 36, 3]
```

Comment:
1. also need to support 18, 34 and 200
2. you can refer backbone in static-graph, use a map